lib/Rest/PathMatch.h

```diff
@@ -0,0 +1,108 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2022-2022 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Tobias Gödderz
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <string>
+#include <string_view>
+#include <tuple>
+#include <utility>
+#include <vector>
+
```

Comment:
```suggestion

/*
 * SYNOPSIS
 * rest::Match helps to match paths. Take for example
 * 
 *   /_api/gharial/{graph}/vertex/{collection}/{vertex}
 *   
 * . This is usually provided as a std::vector<std::string> by
 * GeneralRequest::suffixes(), e.g.
 * 
 *   auto const suffixes = _request->suffixes();
 *   
 * . Note that usually some prefix of the complete path is already stripped, but
 * for the sake of this example, let's assume we got the full path like
 * 
 *   auto const suffixes = std::vector<std::string>{
 *     "_api", "gharial", "MyGraph", "vertex", "SomeCollection", "vertexKey123"
 *   };
 *
 * This can be matched like so:
 * 
 *   if (std::string_view graph, collection, vertex
 *     Match(suffixes).against("_api", "gharial", &graph, "vertex", &collection, &vertex)) {
 *     ... do something with graph, collection, and vertex ...
 *   }
 * 
 * It is always a complete match against the full path. It would be useful to
 * have a prefix-match which also returns the remaining suffix, but it's not
 * implemented (yet)
 * 
 * The argument to Match must be an input range, which elements are convertible
 * to string_views. So e.g. std::vector<std::string> or std::vector<std::string_view>
 * are fine, or other ranges like `suffixes | std::views::drop(2)`.
 * 
 * The arguments of against must either be match-constants or match-variables.
 * A match-constant is something convertible to a std::string_view.
 *   e.g. std::string, std::string_view, or const char*.
 * A match-variable is a pointer to something which is assignable by a std::string_view.
 *   e.g. std::string_view*, or std::string*.
 */

```